### ISSUE TITLE
Pattern doesn't match correctly.

### ISSUE THREAD

**Issue by pufmat** [becheran/wildmatch#18]
**Describe the bug**
Pattern `*113` should match `1113`.

**To Reproduce**
```rs
assert!(WildMatch::new("*113").matches("1113"));
```

**Expected behavior**
Assertion should pass.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Fedora Linux 39 (Workstation Edition)

**Additional context**
`rustc 1.75.0 (82e1608df 2023-12-21)`

**Comment by becheran** [comment]
Fixed with release `2.3.1`

**Comment by pufmat** [comment]
I have just found that this also doesn't pass:
```rs
assert!(WildMatch::new("*121").matches("12121"));
```
